src/utils/index.ts

```diff
@@ -52,3 +55,12 @@ export function hexToBytes(hexString: string): number[] {
   }
   return bytes;
 }
+
+/** @internal */
+const CRYPTO_RESOLVER_ADVANCED_EVENTS_STARTING_BLOCK = 9832516;
+
+export async function getStartingBlock(contract: Contract, tokenId: string) {
+  const logs =  await contract.fetchLogs("ResetRecords", tokenId, 'earliest');
+  const lastResetEvent = logs[logs.length - 1];
+  return lastResetEvent?.blockNumber || CRYPTO_RESOLVER_ADVANCED_EVENTS_STARTING_BLOCK;
```

Comment:
I'd make sure that this function returns the same type in all cases. I have some doubts that `blockNumber` is really a number. Could be either a decimal string or a hex BigInt

Reply:
You are correct, I saw the number as a type for FromBlock over [here](https://github.com/ethers-io/ethers.js/blob/d817416bae2fbc7adb8391fd038613241b7ab8ba/packages/abstract-provider/lib.esm/index.d.ts#L27)
And I am absolutely sure that the tests were passing when I checked them before. My guess is that the blockchain provider might get in the way when you send number rather than a hex string, I have been playing with different providers recently and now my tests are failing when I send number instead of a hex. 

I have changed that back